### ISSUE TITLE
Reload document on application:openFile:

### ISSUE THREAD

**Issue by sveinbjornt** [uranusjr/macdown#576]
I often use MacDown to view Markdown documents that I edit using XCode, by selecting "Open with External Editor" from within XCode. However, the Markdown document is not reloaded when reopened with Macdown.

It would be great if Macdown reloaded the Markdown document when it receives an application:openFile: event. Just a suggestion from a casual user.

**Comment by joelwross** [comment]
+1. Running external tools to modify the file (e.g., doctoc) it would be useful if the source window would update on changes, or if there was a way to "reload" the source without opening and closing the document.

**Comment by franklinyu** [comment]
I tried these steps but didn't reproduce the issue:

1. open `foo.md` with MacDown;
2. close the window of `foo.md`;
3. edit `foo.md` in another editor;
4. open `foo.md` with MacDown, again;

I saw an updated `foo.md`. Did I miss anything?

**Comment by klonos** [comment]
>Did I miss anything?

Yep, step 2: IIUTC you should not close the document from MacDown. If the file is edited with another editor and then saved **while still open in MacDown**, then the document should be reloaded in MacDown (so that the changes from the other editor are loaded).

**Comment by sveinbjornt** [comment]
The best logic for this is:

    if unchanged document open and it is changed by another program:
        update document with changes from the other program
    else if changed document open and it is changed by another program:
        ask user if he wants to overwrite his changes or keep own edits

**Comment by uranusjr** [comment]
> […] when it receives an `application:openFile:` event.

Would this work? I though this is called only when you open a file. This would be the way to go if this is also called when an open document becomes active. 

Note: I did try using `presentedItemDidChange` to detect external modification (#651), but that doesn’t work.

**Comment by sveinbjornt** [comment]
I've used [VDKQueue](https://github.com/bdkjones/VDKQueue) for this kind of thing in my own projects.

**Comment by franklinyu** [comment]
Then this is a duplicate of #48. How does `application:openFile:` help here? Will that be automatically called by the system when the file is updated?

**Comment by franklinyu** [comment]
Yes @uranusjr mentioned VDKQueue in https://github.com/MacDownApp/macdown/pull/564#issuecomment-243976268.

**Comment by sveinbjornt** [comment]
applicationShouldHandleReopen is what I had in mind initially:

https://developer.apple.com/reference/appkit/nsapplicationdelegate/1428638-applicationshouldhandlereopen?language=objc

But VDKQueue is the way to go. That way, the file is automatically updated without MacDown needing an apple event to trigger it.

**Comment by uranusjr** [comment]
The problem about VDKQueue is I can’t find any documentation nor discussion about it. And I make sure it does not have any quirks before using it.

**Comment by sveinbjornt** [comment]
I've used it in my projects for years without trouble, but YMMV.

**Comment by uranusjr** [comment]
It seems like [Transmission also uses it](https://trac.transmissionbt.com/browser/trunk/macosx/VDKQueue/VDKQueue.h?rev=13804)… I guess that’s good enough.

**Comment by franklinyu** [comment]
Note that

> you'll want to modify the code to make it Arc-compliant if you use it in that environment.

according to bdkjones/VDKQueue#8.

**Comment by getaaron** [comment]
Doesn't `NSDocument` handle this by calling `readFromData: ofType: error:`? And in this method we call `[self reloadFromLoadedString]`.

Is there a bug in this implementation? Not sure why we need `VDKQueue` if `NSDocument` handles the file changes for us.
